### ISSUE TITLE
Specifying custom tag names does not work

### ISSUE THREAD

**Issue by thompson-tomo** [dotnet/extensions#5196]
I am trying to use custom names for my properties however that attribute is not working. For reference my log generator is as follows:
```c#
[LoggerMessage(
            EventId = 0,
            Message = "Have recorded metric")]
        public static partial void RecordMetric(
#pragma warning disable EXTEXP0003 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
            this ILogger logger, LogLevel logLevel, [LogProperties(SkipNullProperties = true)] [TagName("test")] MetricDef metric, [LogProperties(SkipNullProperties = true)] MetricDescriptorDef descriptor, [LogProperties(SkipNullProperties = true)] EventDef @event);
#pragma warning restore EXTEXP0003 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.

``` 

The code compiles but the property doesn't change name. to reflect.

Let me know if this needs to become a feature request.

**Comment by joperezr** [comment]
cc: @xakep139

**Comment by mariamgerges** [comment]
@geeknoid is this behavior intended? 
@thompson-tomo can you please provide more clarification, what's your expected behavior? and what's the current behavior?

**Comment by geeknoid** [comment]
Sorry, I didn't see this earlier.

@thompson-tomo if you still remember the issue, could you clarity a bit? Are you expecting that the "metric" parameter shows up as a value named as "test"? Can you show the output of this log statement to see what actually comes out?

**Comment by thompson-tomo** [comment]
That is correct, I am expecting the metric property to be named tag. Will come back in the coming days with the log output etc

**Comment by mariamgerges** [comment]
I tested your code, and created a dummy class for `MetricDef` to see what the output will be, and here's what I got
This is the behavior you're describing:
![Image](https://github.com/user-attachments/assets/bde21a4f-3095-4ba8-a118-1e242b6c2de4)

However, if you add a placeholder in the logging message with the tagName like this "Have recorded metric {test}", you will find the metric property to be "test". 
Here's the behavior when I added the placeholder.
![Image](https://github.com/user-attachments/assets/44cf9cf6-7dc1-4fa5-b476-cb79eb32c5aa)
Is this what you need?
